src/lldp/lldppkt.rs

```diff
@@ -0,0 +1,1086 @@
+                header.r#type,
+            )));
+        }
+
+        let mut payload = buf
+            .get_buffer(header.length.into())
+            .ok_or(pktparser::ParseError::UnexpectedEndOfInput)?;
+
+        let subtype = ChassisIDType::from_wire(&mut payload)?;
+
+        let identifier: Vec<u8> = payload
+            .get_vec(payload.remaining())
+            .ok_or(pktparser::ParseError::UnexpectedEndOfInput)?;
+
+        Ok(ChassisID {
+            r#type: subtype,
+            identifier,
+        })
+    }
+}
+
+impl std::fmt::Display for ChassisID {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "ChassisID type: {} value: {:02x?}",
+            self.r#type, self.identifier
+        )
+    }
+}
+
+#[derive(Debug)]
+pub enum ChassisIDType {
+    ChassisComponent,
+    InterfaceAlias,
+    PortComponent,
+    MacAddress,
+    NetworkAddress,
+    InterfaceName,
+    Local,
+}
+
+impl FromWire for ChassisIDType {
+    fn from_wire(
+        buf: &mut pktparser::Buffer<'_>,
+    ) -> std::result::Result<ChassisIDType, pktparser::ParseError> {
+        Ok(
+            match buf
+                .get_u8()
+                .ok_or(pktparser::ParseError::UnexpectedEndOfInput)?
+            {
+                1 => Self::ChassisComponent,
+                2 => Self::InterfaceAlias,
+                3 => Self::PortComponent,
+                4 => Self::MacAddress,
+                5 => Self::NetworkAddress,
+                6 => Self::InterfaceName,
+                7 => Self::Local,
+                other => {
+                    return Err(pktparser::ParseError::InvalidArgument(format!(
+                        "Unknown ChassisIDType {}",
+                        other
+                    )))
+                }
+            },
+        )
+    }
+}
+
+impl ToWire for ChassisIDType {
+    fn to_wire(&self) -> std::result::Result<Vec<u8>, std::io::Error> {
+        Ok(match self {
+            Self::ChassisComponent => vec![1],
+            Self::InterfaceAlias => vec![2],
+            Self::PortComponent => vec![3],
+            Self::MacAddress => vec![4],
+            Self::NetworkAddress => vec![5],
+            Self::InterfaceName => vec![6],
+            Self::Local => vec![7],
+        })
+    }
+}
+
+impl std::fmt::Display for ChassisIDType {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::result::Result<(), std::fmt::Error> {
+        match self {
+            Self::ChassisComponent => write!(f, "ChassisComponent"),
+            Self::InterfaceAlias => write!(f, "InterfaceAlias"),
+            Self::PortComponent => write!(f, "PortComponent"),
+            Self::MacAddress => write!(f, "MacAddress"),
+            Self::NetworkAddress => write!(f, "NetworkAddress"),
+            Self::InterfaceName => write!(f, "InterfaceName"),
+            Self::Local => write!(f, "Local"),
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct PortID {
```

Comment:
this should be PortId etc, and so on for the various types.

Reply:
Done